src/NLog/Layouts/LayoutParser.cs

```diff
@@ -494,13 +499,11 @@ private static bool CanBeConvertedToLiteral(LayoutRenderer lr)
             return true;
         }
 
-        private static void MergeLiterals(List<LayoutRenderer> list)
+        private static void MergeLiterals(IList<LayoutRenderer> list)
         {
             for (int i = 0; i + 1 < list.Count;)
             {
-                var lr1 = list[i] as LiteralLayoutRenderer;
-                var lr2 = list[i + 1] as LiteralLayoutRenderer;
-                if (lr1 != null && lr2 != null)
+                if (list[i] is LiteralLayoutRenderer lr1 && list[i + 1] is LiteralLayoutRenderer lr2)
                 {
                     lr1.Text += lr2.Text;
```

Comment:
You forgot to reset the ability to return raw-value when merging two literals into one.

Reply:
Well when combining two you haven't a raw value? 

Dunno which test case I'm missing then. (I solved this TDD style)

Reply:
What raw value does this Layout return: `${processid}-${processname}` ?

Reply:
Fixed!